File: Other files/maxSubArraySum.java
class Sum {
    public static int maximum(int[] a) {
        int n = a.length;
        int maxSum = Integer.MIN_VALUE;
        for (int i = 0; i <n ; i++) {
            int current = 0;
            for (int j = i; j < n; j++) {
             current = current + a[j];
            if (current > maxSum) {
                maxSum = current;
            }
            }
        }
        return maxSum;
    }
    public static void main(String args[]) {
        // Your code goes here
        int a[] = {1, 3, 8, -2, 6, -8, 5};
        System.out.println(maximum(a));
    }
}

File: Other files/MaxSubarraySum.cpp
//Time Complexity : O(N)
//Space Complexity : O(1)

#include<bits/stdc++.h>
using namespace std;

int maxSubArraySum(int arr[], int size)
{
  int max_sum = INT_MIN, ending_index = 0;
  
  for(int i=0; i<size; i++)
  {
    ending_index = ending_index + arr[i];
    if(max_sum < ending_index)
    {
      max_sum = ending_index;
    }
    if(ending_index < 0)
    {
      ending_index = 0;
    }
  }
  return max_sum;
}

int main()
{
  int arr[] = {4,-3,4,-1,-5,1,9,-6};
  int n = sizeof(arr)/sizeof(arr[0]);
  
  int ans = maxSubArraySum(arr,n);
  cout << "Maximum Subarray Sum :" << ans;
  return 0;
}
